### PR TITLE
kexec: accept a list of files to be passed to the next kernel

### PR DESCRIPTION
**Pull request by rminnich**
The -extra switch allows inclusion of extra files. The files should not
be huge, as they are pulled into an in-memory archive.

Example usage:
In this example, we are on the console of a node.
cd /
cp /etc/hosts .
cp /dev/kmsg .
kexec -extra 'kmsg hosts' -i initrd.cpio kernel

node reboots ...
~/# ls -l kmsg hosts
-r-------- root 0 24295 Aug 27 15:58 kmsg
-rw-r--r-- root 0 1211 Aug 27 15:58 hosts

The files are available. The main use of this option
at the moment is to preserve the kernel log buffer
for the next kernel.

This is a proof of concept; change at will.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by 10000TB** [comment]
nice,another step forward. One thing worth noting is if/how  to differentiate the dmesg from previous kernel from that from current kernel

File: cmds/core/kexec/kexec_linux.go [hunk 1]
```diff
@@ -40,13 +42,15 @@ type options struct {
...
 	debug        bool
+	extra        string
 	modules      []string
 }
 
 func registerFlags() *options {
 	o := &options{}
 	flag.StringVarP(&o.cmdline, "cmdline", "c", "", "Append to the kernel command line")
 	flag.StringVar(&o.cmdline, "append", "", "Append to the kernel command line")
+	flag.StringVarP(&o.extra, "extra", "x", "", "Add a cpio containing extra files")
```
Comment (10000TB) [inline cmds/core/kexec/kexec_linux.go:53]: typo ? looks like need be empty - per example usage: kexec -extra 'kmsg hosts' -i initrd.cpio kernel

Comment (rminnich) [reply cmds/core/kexec/kexec_linux.go:53]: it is empty. This is StringVarP, not StringVar.

Comment (10000TB) [reply cmds/core/kexec/kexec_linux.go:53]: my bad.. occasionally asking the obvious unnecessarily

**Review by 10000TB — APPROVED** [review]

**Pull request merged**